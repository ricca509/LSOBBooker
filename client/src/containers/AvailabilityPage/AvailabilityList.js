import React from 'react';
import AvailabilityItem from './AvailabilityItem';

const AvailabilityList = ({ availability }) => (
  <table>
    <thead>
      <tr>
        <th>Date</th>
        <th>Times</th>
      </tr>
    </thead>
    <tbody>
      {availability.map(av => <AvailabilityItem
        key={av.date}
        availability={av}
      />)}
    </tbody>
  </table>
);

AvailabilityList.displayName = 'AvailabilityList';

export default AvailabilityList;
