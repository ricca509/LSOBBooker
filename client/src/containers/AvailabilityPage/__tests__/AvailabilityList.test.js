import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import AvailabilityList from '../AvailabilityList';

describe('AvailabilityList', () => {
  const availability = [
    {
      date: "2017-01-03T13:09:55.189",
      times: []
    },
    {
      date: "2017-01-04T13:09:55.189",
      times: [ '12:00:00', '20:00:00' ]
    }
  ];

  it('renders a list of AvailabilityItem', () => {
    const wrapper = shallow(<AvailabilityList availability={availability} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
