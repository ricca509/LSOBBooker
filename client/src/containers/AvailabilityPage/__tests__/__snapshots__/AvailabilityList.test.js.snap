exports[`AvailabilityList renders a list of AvailabilityItem 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Date
      </th>
      <th>
        Times
      </th>
    </tr>
  </thead>
  <tbody>
    <AvailabilityItem
      availability={
        Object {
          "date": "2017-01-03T13:09:55.189",
          "times": Array [],
        }
      } />
    <AvailabilityItem
      availability={
        Object {
          "date": "2017-01-04T13:09:55.189",
          "times": Array [
            "12:00:00",
            "20:00:00",
          ],
        }
      } />
  </tbody>
</table>
`;
